components/org.wso2.carbon.identity.application.authenticator.hypr/src/main/java/org/wso2/carbon/identity/application/authenticator/hypr/HyprAuthenticator.java

```diff
@@ -223,7 +230,23 @@ private void redirectHYPRLoginPage(HttpServletResponse response, AuthenticationC
     private void initiateHYPRAuthenticationRequest(HttpServletRequest request, HttpServletResponse response,
                                                    AuthenticationContext context) throws AuthenticationFailedException {
 
-        String username = request.getParameter(HYPR.USERNAME);
+        String username = null;
+        if (context.getSequenceConfig() != null) {
+            Map<Integer, StepConfig> stepConfigMap = context.getSequenceConfig().getStepMap();
+            // loop through the authentication steps and find the authenticated user from the subject identifier step.
+            if (stepConfigMap != null) {
+                for (StepConfig stepConfig : stepConfigMap.values()) {
+                    if (stepConfig.getAuthenticatedUser() != null && stepConfig.isSubjectIdentifierStep()) {
```

Comment:
I think better to change the order here
first check whether this step is the subject identifier step and then check the authenticated user is not null.

Reply:
addressed with 0bfcecd